pkg/lib/serverTime.js

```diff
@@ -321,13 +309,89 @@ export function ServerTime() {
                     console.warn("failed to load time servers", error);
                     return result;
                 });
-    };
+    }
+
+    function set_custom_ntp_timesyncd(config) {
+        const custom_ntp_config_file = cockpit.file("/etc/systemd/timesyncd.conf.d/50-cockpit.conf",
+                                                    { superuser: "try" });
 
-    self.set_custom_ntp = function (servers, enabled) {
-        const text = `# This file is automatically generated by Cockpit\n\n[Time]\n${enabled ? "" : "#"}NTP=${servers.join(" ")}\n`;
+        const text = `# This file is automatically generated by Cockpit\n\n[Time]\n${config.enabled ? "" : "#"}NTP=${config.servers.join(" ")}\n`;
 
         return cockpit.spawn(["mkdir", "-p", "/etc/systemd/timesyncd.conf.d"], { superuser: "try" })
                 .then(() => custom_ntp_config_file.replace(text));
+    }
+
+    const chronyd_sourcedir = "/etc/chrony/sources.d";
+    const chronyd_sources_enabled = chronyd_sourcedir + "/cockpit.sources";
+    const chronyd_sources_disabled = chronyd_sourcedir + "/cockpit.disabled";
+
+    function get_custom_ntp_chronyd() {
+        const enabled_file = cockpit.file(chronyd_sources_enabled, { superuser: "try" });
+        const disabled_file = cockpit.file(chronyd_sources_disabled, { superuser: "try" });
```

Comment:
Reading shouldn't require superuser privs?

Reply:
Maybe not, but if we have the privs, why not use them?